### DISCUSSION TITLE
Screen Space Curvature Shader NEED HELP PLS

### DISCUSSION THREAD

**Discussion by TerminalTitanium** [xelatihy/yocto-gl#1434] [Q&A]
Hello!

Your library is the best. Pls can anyone help me, how can I add this effect to scene?
<img width="548" alt="image" src="https://github.com/xelatihy/yocto-gl/assets/31609285/c8cf6653-b000-4935-99e4-80ffe7d9e131">

described here
https://madebyevan.com/shaders/curvature/

could you suggest a starting point

**Comment by giacomonazzaro** [comment]
This really depends on the type of shape you want to shade. Is it an implicit surface or a triangle mesh?

**Reply by TerminalTitanium** [reply]
I want to have an effect like in blender Cavity. I work with triangle meshes.
<img width="615" alt="Screenshot 2023-06-16 at 00 55 32" src="https://github.com/xelatihy/yocto-gl/assets/31609285/646e9d4a-2427-42a9-8d94-1c8d76fe8d5e">

**Reply by TerminalTitanium** [reply]
Any ideas? how and where can I apply this shader?

**Reply by TerminalTitanium** [reply]
I still need this option. I would be very grateful if someone offered their services to write this option.

**Comment by TerminalTitanium** [comment]
I found more information how this works.
https://learn.foundry.com/modo/content/help/pages/shading_lighting/shader_items/rt_curvature.html

> It works by analyzing the surface in screen-space (whats visible to the viewer) recovering the normal information and then building the shading sample off the defined thresholds. shader works by applying the gradient values designated in the shader's properties. The outside ends of the gradient terminate in the concave and convex areas of the surfaces topology. The degree of its application can be modified with the Support Size and Magnitude controls.
